var/spack/repos/builtin/packages/ImageMagick/package.py

```diff
@@ -42,6 +43,9 @@ class Imagemagick(Package):
     depends_on('libtiff')
     depends_on('ghostscript')
 
+    def url_for_version(self, version):
+        return "https://github.com/ImageMagick/ImageMagick/archive/{0}.tar.gz".format(version)
```

Comment:
This should work automatically, but the version was getting mangled. I think it must be due to the dash in the version. Adding the url_for_version solves the problem.